src/core/UBPreferencesController.cpp

```diff
@@ -52,19 +52,19 @@ qreal UBPreferencesController::sSliderRatio = 10.0;
 qreal UBPreferencesController::sMinPenWidth = 0.5;
 qreal UBPreferencesController::sMaxPenWidth = 50.0;
 
-// convenience function to convert between screen list as QString and QStringList
-QStringList splitScreenList(const QString& input)
+// convenience function to split a comma separated list of tokens to a QStringList
+QStringList trimmedTokens(const QString& input)
```

Comment:
I (personnally) prefer when function and variable names are explicit on what it represents in functional terms rather than in a technical aspect. Maybe something like `getScreenNamesFromInput` or `toScreenNames` or `convertToScreenNames` with the idea to have something like `auto screenNames = extractScreenNames(input); for (auto screeName : screenNames) ...` when using it. 

for example, I would expect from a function like `splitScreenList` to have a signature like `QList<QScreen*> splitScreenList(const QString& input)`. For `trimmedTokens`, I would expect something like `void trimmedTokens(const QStringList& tokens)`

edit : `screenIndexes` is probably even more accurate

Reply:
> I (personnally) prefer when function and variable names are explicit on what it represents in functional terms rather than in a technical aspect.

Here I might have the opposite (personal) preference ;) My argument would be that a function operating on a string and returning a somehow splitted version of it has no idea that the content of that string has something to do with screens and should therefore not have "screen" in its name. It is the **invoker** who uses it for the purpose of splitting a screen index list. The function could also be used to split a list of domain names, options, or whatever in another context. Just think you move this function out of this source file to some common string utility place. Then it would be strange if it still contained "screen" in its name. And indeed, would perhaps `UBTextTools` also be a good location for this function?

In the end the function mentioned above is some specialized version of `QString::split`, so I could imagine things like
- `splitAndTrim`
- `splitStringAndTrim`
- `splitToTrimmedTokens`

Reply:
Oh, I totally agree IF the function is generic but then, indeed its location should be in a class responsible of manipulating strings like `UBStringUtils` in the frameworks folder.

Now I understand that you were trying to make your new classes reusable, like with UBStringListValidator. It should also be located in `UBStringUtils`, imo.

If the intent is to make these functions more generic, I think it could also respect SRP. My experience is that when I need to put an `and` in a function's name, either I didn't find the correct word explaining its responsibility, or I should have make two functions. 

Maybe having something like `static QStringList UBStringUtils::toStringList(const QString& source, const QChar separator) ` and `static QStringList UBStringUtils::trim(const QStringList& list)` would be more accurate and reusable (and it would respect SRP). What do you think ?

Reply:
Very good idea! Agreed!

Reply:
In my opinions everybody wants to be generic and wants to make things reusable. But then the situation never comes, where something is actually reused. UBPreferenceController is no good location to define such general functions, and it is no good location to search for such general functions. If the UBStringListValidator was really something that is worth having it in general, it would be part of Qt or of some utility library.
Therefore I always prefer to use names that represent functional aspects. This enables an abstract way of programming. And this results in code that is nice to read and to understand. Code is typically written once, but read multiple times. Technical aspects should not be mixed with functional aspects and should not distract from them. Using strings is just an unimportant technical detail and therefore should not influence how you name your functions. Someone may even question if using strings is a good idea. In my opinion it does not matter if you use strings or not. If you change the internal data representation to something else, the code should not have to be changed, because the technical details should be encapsulated away from your code.

That is just my opinion. But I always respect the decion of the author.

Reply:
See commit https://github.com/OpenBoard-org/OpenBoard/pull/633/commits/8d5d76a6a56a7e50cbef1acbe0cb6655eacddfd5 as the result of the discussion above. 

Still the `UBStringListValidator` is part of the preferences controller. You are right, one would not expect a reusable class here. For me however there is another reason for making classes in a reusable way: it typically also improves "separation of concerns", and this is an argument even if in the end nobody actually reuses the code.